### PR TITLE
Basic components

### PR DESCRIPTION
**Pull request by SomeoneWithBFG**

### CODE REVIEW

File: react-webapp/src/components/complex/Menu/index.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      <MenuUserItem 
+        name={"Иванов И.И."}
+        role={"Студент"}
+        group={"АС-53"}
+        link={"/students?id=some-id"}
+      />
+      {pages.map(page=>{
+        return <MenuItem key={page.title} props={page}/>
+      })}
+    </div>
```
Comment (slavasgod) [inline react-webapp/src/components/complex/Menu/index.tsx:27]: See comment above

File: react-webapp/src/routes/index.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+        <Route
+            path="/sandbox"
+            component={Sandbox}
+        />
+   </Switch>
+  )
+  
+  export default withRouter(GeneralRouterComponent)
+
```
Comment (slavasgod) [inline react-webapp/src/routes/index.tsx:24]: 1. Add empty line
2. Why do you use `withRouter` here?

File: react-webapp/src/styles/colors/colors.scss [hunk 3]
```diff
@@ -1,8 +1,23 @@
...
+$cyan: #13C2C2;
+$blue: #1890FF;
+$geekblue: #2F54EB;
+$purple: #722ED1;
+$magenta: #EB2F96;
+
+$very-light-gray: #f5f6f7;
+$light-gray: #d6d5d5;
+$little-light-gray: #bdbdbd;
+$gray: #666666;
```
Comment (slavasgod) [inline react-webapp/src/styles/colors/colors.scss:23]: Empty line

File: react-webapp/src/styles/fonts/fonts.scss [hunk 4]
```diff
@@ -0,0 +1 @@
+$basic: 'Nunito', 'sans-serif';
```
Comment (slavasgod) [inline react-webapp/src/styles/fonts/fonts.scss:1]: Empty line

File: react-webapp/src/useCases/services/HistoryService.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  }
+
+  push(route: Route) {
+    this.history.push(route)
+  }
+
+  back() {
+    this.history.goBack()
+  }
+}
```
Comment (slavasgod) [inline react-webapp/src/useCases/services/HistoryService.ts:22]: Not sure that we need this, but ok

File: react-webapp/src/pages/Arcticles/Articles.tsx [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+      <div style={{margin:"1rem"}}>
+      <Input 
+        onChange={testFunction}
+      />
+      </div>
+
+      <div style={{margin:"1rem"}}>
+      <Button 
+        onChange={testFunction}
+      />
```
Comment (slavasgod) [inline react-webapp/src/pages/Arcticles/Articles.tsx:62]: When you test something like this, remove it before push

**Review by slavasgod — CHANGES_REQUESTED** [review]

File: react-webapp/src/components/common/MenuUserItem/index.tsx [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    link: string,
+}
+
+const MenuUserItem: FC<MenuUserItemProps> = (props) => {
+    const data = {
+        name: props.name.length > 20 ? props.name.slice(-20)+"..." : props.name,
+        role: props.role,
+        group: props.group ? props.group : null,
+        link: props.link,
+    }
```
[the remark this answers is not in the document]
Comment (SomeoneWithBFG) [reply react-webapp/src/components/common/MenuUserItem/index.tsx:19]: text overflow

File: react-webapp/src/components/common/MenuUserItem/index.tsx [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
+interface MenuUserItemProps {
+    name: string,
+    role: string,
+    group?: string,
+    link: string,
+}
+
+const MenuUserItem: FC<MenuUserItemProps> = (props) => {
+    return (
```
Comment (slavasgod) [inline react-webapp/src/components/common/MenuUserItem/index.tsx:14]: Use object spread to props

File: react-webapp/src/components/complex/Menu/pages.ts [hunk 9]
```diff
@@ -0,0 +1,7 @@
+export const pages = [
+    {title: "Вычисления", link: "/"},
+    {title: "История",    link: "/history"},
+    {title: "Группа",     link: "/group"},
+    {title: "Студенты",   link: "/students"},
+    {title: "Обучение",   link: "/stydy"},
+]
```
Comment (slavasgod) [inline react-webapp/src/components/complex/Menu/pages.ts:7]: Empty line

File: react-webapp/src/routes/index.tsx [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+        <Route
+            path="/sandbox"
+            component={Sandbox}
+        />
+   </Switch>
+  )
+  
+  export default GeneralRouterComponent
+
```
Comment (slavasgod) [inline react-webapp/src/routes/index.tsx:23]: empty line

**Review by slavasgod — CHANGES_REQUESTED** [review]

**Review by slavasgod — APPROVED** [review]

**Pull request merged**